src/main/java/net/snowflake/client/core/HttpClientSettingsKey.java

```diff
@@ -124,16 +123,24 @@ public String getUserAgentSuffix() {
   }
 
   /** Be careful of using this! Should only be called when password is later masked. */
-  String getProxyPassword() {
+  public String getProxyPassword() {
     return this.proxyPassword;
   }
 
   public String getNonProxyHosts() {
     return this.nonProxyHosts;
   }
 
-  public Protocol getProxyProtocol() {
-    return this.proxyProtocol.equalsIgnoreCase("https") ? Protocol.HTTPS : Protocol.HTTP;
+  /**
+   * @deprecated Use {@link #getProxyHttpProtocol()}
+   * @return ProxyProtocol
+   */
+  @Deprecated
+  public com.amazonaws.Protocol getProxyProtocol() {
+    return this.proxyProtocol.equalsIgnoreCase("https") ? com.amazonaws.Protocol.HTTPS : com.amazonaws.Protocol.HTTP;
+  }
+  public HttpProtocol getProxyHttpProtocol() {
```

Comment:
remember to run formatter as described in https://github.com/snowflakedb/snowflake-jdbc/?tab=readme-ov-file#development - check style is our required job